### DISCUSSION TITLE
Extend only part of the component

### DISCUSSION THREAD

**Discussion by milelazar** [nuxt/nuxt#16150] [Questions] [answered]
I have a central "nuxt-base" project with custom component library shared across multiple other project. Some components share the same template/HTML and script logic across all projects. Only css needs to be different. Is there a way to override the css without duplicating whole component to every project? This kind of beats the purpose of base components library.

I could override the styles in global css file but this is not ideal. I would like to have it scoped per component.
Not sure if possible but it would be great if you could create the same component but extend/override only desired section.

For example

/nuxt-base/components/component.vue
```
<template>
<div>Component template</div>
</template>

<script>
...component js logic...
</script>

<style scoped>
...component css styles...
</style>
```

/custom-project/components/component.vue
Other project components just need to have different css. Script and template sections should be extended from the base project
```
<template>
Use base component template
</template>

<script>
Use base component script
</script>

<style scoped>
...override css styles...
</style>
```

**Comment by IsraelOrtuno** [comment, marked as answer]
You could just use a wrapper component. Basically create a new component which will use the base component but overwrite the styles, a basic example could look like this 👇

```vue
<template>
  <BaseComponent v-bind="$attrs" />
</template>

<style>
// Your styles here
</style>
```

**Reply by yuriystasiv** [reply]
> You could just use a wrapper component. Basically create a new component which will use the base component but overwrite the styles, a basic example could look like this 👇
> 
> ```vue
> <template>
>   <BaseComponent v-bind="$attrs" />
> </template>
> 
> <style>
> // Your styles here
> </style>
> ```

Looks like this is the only way to override styles when using Nuxt Layers, right @danielroe?
How did you end up @milelazar?

**Reply by milelazar** [reply]
@yuriystasiv I ended up using the wrapper component. Works fine for me. But you need to use ":deep" css selector if you want to override styles within extended component

**Comment by memic84** [comment]
Do you need to override the whole style section, or do you have also some base styles set?

I believe that you could also reference the style in your base component:

```vue
<style src="./component.scss" lang="scss" scoped></style>
```

Haven't used this myself, we're using Nuxt 3 config extends with tailwind (tailwind.config is merged between base & custom) and that's enough for us with other themes, where you still have the full flexibility if you need to change the template / html for one component.

**Comment by milelazar** [comment]
Thanks guys. Both answers are very helpfull. Will see which option will work better for me down the road.
